spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/functions/SparkFunctions.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.spark.functions;
+
+import java.util.Locale;
+import java.util.Map;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.spark.sql.connector.catalog.functions.UnboundFunction;
+
+public class SparkFunctions {
+
+  private static final Map<String, UnboundFunction> FUNCTIONS = initFunctions();
+
+  private SparkFunctions() {
+  }
+
+  // The caller needs to validate that the namespace exists and is an Iceberg namespace when using these results.
+  //
+  // Functions that are added to all Iceberg catalogs and their associated namespaces get registered
+  // under Namespace.empty(), allowing the calling code to add the namespace.
+  public static String[] list() {
```

Comment:
Why does this return an array?

If it were a list, this could produce an `ImmutableList` and make that a static field, rather than getting the key set and converting it to an array every call.

Reply:
Made it an immutable list and converted to a static field.